fileshare/event_manager.go

```diff
@@ -206,6 +240,29 @@ func (em *EventManager) handleRequestReceivedEvent(event EventKindRequestReceive
 	}
 }
 
+func (em *EventManager) withProgressCh(transferID string, fn func(ch chan TransferProgressInfo)) {
+	if ch, ok := em.transferSubscriptions[transferID]; ok {
+		fn(ch)
+	}
+}
+
+func (em *EventManager) reportProgress(transferID string, status pb.Status, transferred uint32) {
+	em.withProgressCh(transferID, func(ch chan TransferProgressInfo) {
+		progress := TransferProgressInfo{
+			TransferID:  transferID,
+			Transferred: transferred,
+			Status:      status,
+		}
+		select {
+		case ch <- progress:
+		default:
+			log.Println(internal.WarningPrefix, " progress channel is full. removing oldest item and sending")
+			<-ch
```

Comment:
So this will drop progress events in case the channel is full. I assume this will happen only when we are too slow with processing events from libdrop right?

Reply:
yes, that's right. I made an assumption that the progress events are not that important, and dropping an event is not a big deal, because you'll have other 32 events waiting for you anyways.